src/js/elements/data_mapper/components/richDataPanelLink.js

```diff
@@ -0,0 +1,44 @@
+import React, {useCallback} from 'react';
+import {Component} from "../../../utils";
+
+import { ThemeProvider } from '@mui/system';
+import { useTheme } from '@mui/material/styles';
+import {
+    IconContainer, ParentContainer, TextContainer,
+    Link, PanelIconSidebar,  PanelIconLink
+} from './styledElements';
+
+
+const RichDataPanelLink = (props) => {
+  const theme = useTheme();
+
+  const openRichDataPanel = useCallback(
+    (event) => {
+      props.parent.triggerEvent("open.rich_data.panel");
+    },
+    [
+      props.parent
+    ]
+  )
+
+  return (
+    <ThemeProvider theme={theme}>
+      <ParentContainer>
+        <IconContainer>
+          <PanelIconSidebar />
+        </IconContainer>
+        <TextContainer>
```

Comment:
So this is like Reacts version of CSS class

Reply:
yes it is MUI theme based custom react components.
Regarding reusable data factory: We should divide that into two separate tasks
1) Spike to check available and most compatible factory library
2) Implementation of that library and removing all json files.

I think this one deserves epic of it's own